python/ray/train/tests/test_session.py

```diff
@@ -91,8 +90,8 @@ def train_func():
     assert session.get_next().data["loss"] == 1
     shutdown_session()
```

Comment:
Maybe conversation for later, but if `train` methods no longer need to be called inside a session, should we still start a session for this test?

Reply:
I did some refactoring here to move the "outside of session" assertions to a different test.

We still want to test in the session though because the behavior is different.